keps/sig-cli/3655-kubectl-translation-improvement/README.md

```diff
@@ -0,0 +1,1226 @@
+
+TODO: 
+* Do both the `k8s.po` and `k8s.mo` files need to be embedded in kubectl, or can we only embed `k8s.mo`?  We should figure this out and not embed `k8s.po` unless necessary.
+
+<</[UNRESOLVED]>>
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### 1. DETAILS: Make it simple for people to update translations.
+
+Currently, translation contributors need to run `hack/update-translations.sh -x` to extract all strings wrapped in `i18n.T()` into a file called [`template.pot`](https://github.com/kubernetes/kubernetes/blob/33e6ebc8f8df47864a77c867d78216adb70cd79d/staging/src/k8s.io/kubectl/pkg/util/i18n/translations/kubectl/template.pot). This template can be synchronized with a locale's `k8s.po` file (using poedit for example) to bring in new text that needs to be translated.
+
+The `hack/update-translations.sh` script has some prerequisites that must be installed and are unrelated to the job that the translation contributor is there to do. The usage of this script is documented, but not well enough for a first time contributor to be confident in running it.
+
+Ideally however, translation contributors should not need to run any scripts or install any tools, other than a translation editor of their choice to update translations (like poedit).
+
+To make sure that text is available to be translated without the translation contributor needing to run any additional processes, we will require developers to update the translation files as a presubmit condition of their pull request. This will be done through the use of a new script `hack/verify-translations.sh` which will fail if it detects out-of-date or missing translations. In this case, the developer will need to run `hack/update-translations.sh -x -k` to make sure the translation files are up-to-date.
+
+As part of this effort, we should simplify `hack/update-translations.sh` so that you don't need to pass in flags and just make it work for the most common scenario by default (which I think is `-x -k`).
+
+Alternative for consideration:
+* Instead of having a presubmit job and requiring developers to keep the translation files up-to-date, we could run `hack/update-translations.sh` in a postsubmit job and commit any changes that occur back to the k/k repo (is that even possible?), or automatically open a pull request (more likely this will be the way).
```

Comment:
Agreed, I'm leaning that way myself.

I like the idea of incorporating it into `make update`.

`update-translations.sh` has some prerequisites though that only really make sense if you are doing cli work, so we would need to make sure this doesn't burden the other k8s contributors who don't need to worry about it, but if we can figure out a way to do that, it would be great.